sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,134 @@
+        connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.DIRECT);
+        directClient = cosmosClientBuilder.setConnectionPolicy(connectionPolicy).buildClient();
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(this.gatewayClient.asyncClient());
+        container = gatewayClient.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() throws CosmosClientException {
+        assertThat(this.gatewayClient).isNotNull();
+        this.gatewayClient.close();
+        if (this.directClient != null) {
+            this.directClient.close();
+        }
+    }
+
+    @Test(groups = {"simple"})
+    public void gatewayDiagnostics() throws CosmosClientException {
+        CosmosItemProperties cosmosItemProperties = getCosmosItemProperties();
+        CosmosItemResponse createResponse = container.createItem(cosmosItemProperties);
+        String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
+        assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.GATEWAY);
+        assertThat(diagnostics).contains("Gateway statistics");
+        assertThat(diagnostics).contains("Operation Type : " + OperationType.Create);
+        assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+    }
+
+    @Test(groups = {"simple"})
+    public void gatewayDiagnosticsOnException() throws CosmosClientException {
+        CosmosItemProperties cosmosItemProperties = getCosmosItemProperties();
+        CosmosItemResponse createResponse = null;
+        try {
+            createResponse = this.container.createItem(cosmosItemProperties);
+            CosmosItemRequestOptions cosmosItemRequestOptions = new CosmosItemRequestOptions();
+            cosmosItemRequestOptions.setPartitionKey(new PartitionKey("wrongPartitionKey"));
+            CosmosItemResponse readResponse = this.container.getItem(createResponse.getItem().getId(), null).read(cosmosItemRequestOptions);
+            fail("request should fail as partition key is wrong");
+        } catch (CosmosClientException exception) {
+            String diagnostics = exception.getCosmosResponseDiagnostics().toString();
+            assertThat(exception.getStatusCode()).isEqualTo(HttpConstants.StatusCodes.NOTFOUND);
+            assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.GATEWAY);
+            assertThat(diagnostics).contains("Gateway statistics");
+            assertThat(diagnostics).contains("Status Code : 404");
+            assertThat(diagnostics).contains("Operation Type : " + OperationType.Read);
+            assertThat(exception.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+            System.out.println(diagnostics);
```

Comment:
please use logger in the test no system.out.

Reply:
removed it , it was not intentional